gwctl/pkg/printer/backends_test.go

```diff
@@ -0,0 +1,485 @@
+								BackendRef: gatewayv1.BackendRef{
+									BackendObjectReference: gatewayv1.BackendObjectReference{
+										Port:      ptr.To(gatewayv1.PortNumber(8082)),
+										Name:      "foo-svc-2",
+										Kind:      ptr.To(gatewayv1.Kind("Service")),
+										Namespace: ptr.To(gatewayv1.Namespace("ns2")),
+									},
+								},
+							},
+							{
+								BackendRef: gatewayv1.BackendRef{
+									BackendObjectReference: gatewayv1.BackendObjectReference{
+										Port:      ptr.To(gatewayv1.PortNumber(8083)),
+										Name:      "foo-svc-3",
+										Kind:      ptr.To(gatewayv1.Kind("Service")),
+										Namespace: ptr.To(gatewayv1.Namespace("ns3")),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		&gatewayv1.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bar-route-21",
+				Namespace: "ns1",
+				CreationTimestamp: metav1.Time{
+					Time: fakeClock.Now().Add(-9 * time.Hour),
+				},
+			},
+			Spec: gatewayv1.HTTPRouteSpec{
+				Hostnames: []gatewayv1.Hostname{"foo.com", "bar.com", "example.com", "example2.com", "example3.com", "example4.com", "example5.com"},
+				CommonRouteSpec: gatewayv1.CommonRouteSpec{
+					ParentRefs: []gatewayv1.ParentReference{
+						{
+							Kind:      common.PtrTo(gatewayv1.Kind("Gateway")),
+							Group:     common.PtrTo(gatewayv1.Group("gateway.networking.k8s.io")),
+							Namespace: common.PtrTo(gatewayv1.Namespace("default")),
+							Name:      "demo-gateway-2",
+						},
+					},
+				},
+				Rules: []gatewayv1.HTTPRouteRule{
+					{
+						BackendRefs: []gatewayv1.HTTPBackendRef{
+							{
+								BackendRef: gatewayv1.BackendRef{
+									BackendObjectReference: gatewayv1.BackendObjectReference{
+										Port:      ptr.To(gatewayv1.PortNumber(8082)),
+										Name:      "foo-svc-2",
+										Kind:      ptr.To(gatewayv1.Kind("Service")),
+										Namespace: ptr.To(gatewayv1.Namespace("ns2")),
+									},
+								},
+							},
+							{
+								BackendRef: gatewayv1.BackendRef{
+									BackendObjectReference: gatewayv1.BackendObjectReference{
+										Port:      ptr.To(gatewayv1.PortNumber(8083)),
+										Name:      "foo-svc-3",
+										Kind:      ptr.To(gatewayv1.Kind("Service")),
+										Namespace: ptr.To(gatewayv1.Namespace("ns3")),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	finalObjects := []runtime.Object{}
+	finalObjects = append(finalObjects, healthCheckPolicies...)
+	finalObjects = append(finalObjects, timeoutPolicies...)
+	finalObjects = append(finalObjects, objects...)
+
+	params := utils.MustParamsForTest(t, common.MustClientsForTest(t, finalObjects...))
+	discoverer := resourcediscovery.Discoverer{
+		K8sClients:    params.K8sClients,
+		PolicyManager: params.PolicyManager,
+	}
+	resourceModel, err := discoverer.DiscoverResourcesForBackend(resourcediscovery.Filter{})
+	if err != nil {
+		t.Fatalf("Failed to construct resourceModel: %v", resourceModel)
```

Comment:
```suggestion
		t.Fatalf("Failed to construct resourceModel: %v", err)
```

(I know we have the same issue in other places, and we'll get to it independently, or when we  make changes to those tests)